src/lights/DirectionalLight.js

```diff
@@ -28,11 +28,18 @@ DirectionalLight.prototype = Object.assign( Object.create( Light.prototype ), {
 
 	isDirectionalLight: true,
 
-	copy: function ( source ) {
+	copy: function ( source, recursive, cache ) {
 
-		Light.prototype.copy.call( this, source );
+		if ( cache === undefined ) cache = {};
 
-		this.target = source.target.clone();
+		Light.prototype.copy.call( this, source, recursive, cache );
+
+		if ( ( source.target.uuid in cache ) === false ) {
+
+			cache[ source.target.uuid ] = source.target.clone( recursive, cache );
```

Comment:
Here, the cache will only be updated with the target _after_ cloning the target subtree. This means other references (e.g. other lights pointing at the same target) in the target subtree will not use the cached version. If I understand right.